hack/generator/pkg/codegen/pipeline_create_storage_types.go

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// createStorageTypes returns a pipeline stage that creates dedicated storage types for each resource and nested object.
+// Storage versions are created for *all* API versions to allow users of older versions of the operator to easily
+// upgrade. This is of course a bit odd for the first release, but defining the approach from day one is useful.
+func createStorageTypes() PipelineStage {
+	return PipelineStage{
+		id:          "createStorage",
+		description: "Create storage versions of CRD types",
+		Action: func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
+
+			storageTypes := make(astmodel.Types)
+			visitor := makeStorageTypesVisitor(types)
+			vc := makeStorageTypesVisitorContext()
+			var errs []error
+			for _, d := range types {
+				d := d
+
+				if types.IsArmDefinition(&d) {
+					// Skip ARM definitions, we don't need to create storage variants of those
+					continue
+				}
+
+				if _, ok := types.ResolveEnumDefinition(&d); ok {
+					// Skip Enum definitions as we use the base type for storage
+					continue
+				}
+
+				def, err := visitor.VisitDefinition(d, vc)
+				if err != nil {
+					errs = append(errs, err)
+					continue
+				}
+
+				finalDef := def.WithDescription(descriptionForStorageVariant(d))
+				storageTypes[finalDef.Name()] = finalDef
+			}
+
+			if len(errs) > 0 {
+				err := kerrors.NewAggregate(errs)
+				return nil, err
+			}
+
+			types.AddTypes(storageTypes)
+
+			return types, nil
+		},
+	}
+}
+
+// makeStorageTypesVisitor returns a TypeVisitor to do the creation of dedicated storage types
+func makeStorageTypesVisitor(types astmodel.Types) astmodel.TypeVisitor {
+	factory := &StorageTypeFactory{
+		types: types,
+	}
+
+	result := astmodel.MakeTypeVisitor()
+	result.VisitTypeName = factory.visitTypeName
+	result.VisitObjectType = factory.visitObjectType
+	result.VisitArmType = factory.visitArmType
+
+	factory.visitor = result
+	factory.propertyConversions = []propertyConversion{
+		factory.kubernetesResourceStoragePropertyConversion,
+		factory.defaultStoragePropertyConversion,
```

Comment:
These would read better as verb phrases:
```suggestion
		factory.convertKubernetesResourceStorageProperty,
		factory.convertDefaultStorageProperty,
```

Reply:
I used different names, but did make them into verb phrases.